src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,110 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface NewMovieProps {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<NewMovieProps> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [isFormValid, setIsFormValid] = useState(false);
+
+  // простой переменной
+  useEffect(() => {
+    const isTitleValid = title.trim() !== '';
```

Comment:
```suggestion
    const isTitleValid = !!title.trim();
```
or 
```suggestion
    const isTitleValid = Boolean(title.trim());
```
empty string is a falsy value - so you can transform it to boolean instead of comparing with empty string